IanGRap/Squish
# Dash cooldown feedback

Contributors: Coulter

Goal: Add visible feedback that indicates when the dash cooldown ends for a player after dashing.

Files Changed:
- PlayerDash.cs: Within the dash() function, the player's material color is darkened and their point light is disabled after dashing. They remain this way for the duration of the cooldown, and then return to normal to indicate that a player can now dash again. An OriginalColor public property has been added so that the default colors for each player material can be set and stored.
- PlayerColorController.cs: Added lines to set the correct original and cooldown colors in the PlayerDash script. Works only for Sprint 6 scene.
- PlayerSelectColorController.cs: Added lines to set the correct original and cooldown colors in the PlayerDash script. Works only for the lobby scene.
- PlayerRespawner.cs: Added lines to set the correct original and cooldown colors in the PlayerDash script for respawned players.
- SquishDetector.cs: Added a line that resets a player's material right before they die, so that the correct material is passed to the camera splat, respawn block and UI indicator.

How to test: 
- Start a game in either the lobby or Sprint 6 scene.
- Dash. The player point light should disappear and their color should darken.
- Observe whether or not the point light and color are correctly reset once the dash cooldown ends. Both should just return to their original states.
- Attempt to dash right before you get squished, so that your color is darker when you die. The goo splat, respawn UI, and respawn block color should all be the original color.
- Do the same thing again, and successfully respawn. The respawned player color should be correct.

Notes:
It seems that Unity does not like it when you directly change the color of a material. Doing so causes new instances of the material to be made each time. On top of being the source of the major bug with the respawn colors, this also takes up unnecessary memory. The better approaches are to either change the color through shaders, or have separate materials for each color. I'm now using the second approach, wherein two materials are set in the PlayerColorController and PlayerSelectColorController scripts, and these two materials are swapped out in the skinnedmeshrenderer component as needed. It's ugly, but it's the only way I've been able to make it work.
